Fix #11: Sender specific scheduling

Alterrien opened the pull request:
It should work, but it would be nice if @aubincleme ran it on the test mailing list just to be sure.

Commit:
sender-specific scheduling

Commit:
Sender-specific scheduling, second draft

Commit:
Bug fixes

aubincleme commented:
Ok, we could try this during this weekend, or next week :)

Thooms commented on senders/senders.py:
```diff
@@ -39,8 +47,12 @@ def slack_message(self, providers):
-    def __init__(self, fobj=sys.stdout):
+    def __init__(self, ev, unit, fobj=sys.stdout):
+        if unit not in ['seconds', 'minutes', 'hours', 'days']:
+            raise Exception('Bad scheduling information provided')
```
You might wanna factorize this check in the `Sender` class

Thooms commented on reminder/reminder.py:
```diff
@@ -2,35 +2,34 @@
-
-        ev, unit = self.frequence
-
+    def schedule(self,r,ev,unit):
```
Spaces around the arguments :)

Thooms commented:
@Kirogo some comments on the code, otherwise the design seems nice!
Would you add some unit tests :) ? The design of the overall project has come to something nontrivial, we could use those.

Commit:
Code factorisation in senders/senders.py
Fixed spacing in reminder/reminder.py
Added non-daemon method test_run for test purpose in reminder/reminer.py

Alterrien commented:
@Thooms I'm okay for some unit testing, but I'll need some help or documentation on how to do proper unit testing !

Commit:
Fixed example in README.md

Thooms commented on senders/senders.py:
```diff
@@ -9,8 +9,16 @@ class Sender:
 class MailSender(Sender):
-    def __init__ (self, smtp_server, sender_address, recipient_address):
+    def __init__ (self, ev, unit, smtp_server, sender_address, recipient_address):
+        self.scheduleCheck(ev,unit)
```
Add the correct `__init__` function in the `Sender` class, and use the [`super()`](https://docs.python.org/3/library/functions.html#super) builtin instead of your `scheduleCheck` function :) (the exact syntax will be something like `super().__init__(args)`).

Thooms commented on reminder/reminder.py:
```diff
@@ -41,3 +40,16 @@ def main_loop():
         daemon = Daemonize(app='issues-reminder', pid=pidfile, action=main_loop)
         daemon.start()
+
+    def test_run(self):
```
What is this for?

Alterrien replied:
Just a method to test it in a terminal. I can remove it if needed.

Commit:
Code refactoring

Alterrien commented:
Bumping the PR @Thooms
